Modules/Scripted/ScreenCapture/ScreenCapture.py

```diff
@@ -1050,10 +1068,13 @@ def captureImageFromView(self, view, filename, transparentBackground=False):
       imageClipper.Update()
       capturedImage = imageClipper.GetOutput()
 
-    writer = self.createImageWriter(filename)
-    writer.SetInputData(self.addWatermark(capturedImage))
-    writer.SetFileName(filename)
-    writer.Write()
+    if isinstance(volumeNode, slicer.vtkMRMLVectorVolumeNode):
+      volumeNode.SetAndObserveImageData(capturedImage)
+    if filename:
+      writer = self.createImageWriter(filename)
+      writer.SetInputData(self.addWatermark(capturedImage))
```

Comment:
Move `capturedImage = self.addWatermark(capturedImage)` before starting to write to volume node and file, so that the same content is written to both outputs.